Framework/CMakeLists.txt

```diff
@@ -57,6 +57,10 @@ if ( ${CMAKE_SYSTEM_NAME} MATCHES "Linux" )
   endif ( MPI_BUILD )
 endif ()
 
+if ( MPI_EXPERIMENTAL )
```

Comment:
Long-term I don't think that `MPI_EXPERIMENTAL` is what we would want to call it. Could we try to select a suitable flag name now. `MPI_ALGORITHMS` etc?